pkg/build/builder/common.go

```diff
@@ -182,15 +184,29 @@ func updateBuildRevision(build *buildapi.Build, sourceInfo *git.SourceInfo) *bui
 	}
 }
 
-func retryBuildStatusUpdate(build *buildapi.Build, client client.BuildInterface, sourceRev *buildapi.SourceRevision) error {
-	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+// handleBuildStatusUpdate handles updating the build status
+// retries occur on update conflict and unreachable api server
+func handleBuildStatusUpdate(build *buildapi.Build, client client.BuildInterface, sourceRev *buildapi.SourceRevision) {
+	var latestBuild *buildapi.Build
+
+	updateBackoff := wait.Backoff{
+		Steps:    10,
+		Duration: 25 * time.Millisecond,
+		Factor:   2.0,
+		Jitter:   0.1,
+	}
+
+	if updateErr := wait.ExponentialBackoff(updateBackoff, func() (bool, error) {
```

Comment:
really minor nit ... I would just assign updateErr here and then do the if as a separate statement in line 235. To me, it's easier to follow.